### PR TITLE
Add e2e tests for REST errors high alerts

### PR DESCRIPTION
**Pull request by machadovilaca**
Signed-off-by: João Vilaça <jvilaca@redhat.com>

**What this PR does / why we need it**:

Add e2e tests for REST errors high alerts:

- VirtApiRESTErrorsHigh
- VirtControllerRESTErrorsHigh
- VirtHandlerRESTErrorsHigh
- VirtOperatorRESTErrorsHigh

jira-ticket: CNV-22403

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: tests/monitoring/monitoring.go [hunk 1]
```diff
@@ -531,42 +538,48 @@ var _ = Describe("[Serial][sig-monitoring]Prometheus Alerts", Serial, func() {
...
 				}
 			}
 
 			verifyAlertExist(virtApi.restErrorsBurtsAlert)
+			verifyAlertExist(virtApi.restErrorsHighAlert)
 		})
 
-		It("VirtOperatorRESTErrorsBurst should be triggered when requests to virt-operator are failing", func() {
+		It("VirtOperatorRESTErrorsBurst and VirtOperatorRESTErrorsHigh should be triggered when requests to virt-operator are failing", func() {
+			revertScale(virtOperator.deploymentName)
```
Comment (0xFelix) [inline tests/monitoring/monitoring.go:563]: This is already part of `AfterEach`?

File: tests/monitoring/monitoring.go [hunk 2]
```diff
@@ -504,14 +509,16 @@ var _ = Describe("[Serial][sig-monitoring]Prometheus Alerts", Serial, func() {
...
-
 			crb, err = virtClient.RbacV1().ClusterRoleBindings().Get(context.Background(), "kubevirt-operator", metav1.GetOptions{})
 			util.PanicOnError(err)
 
-			reduceAlertPendingTime()
 			increaseRateLimit()
+
+			scales = make(map[string]*autoscalingv1.Scale, 1)
+			backupScale(virtOperator.deploymentName)
+			updateScale(virtOperator.deploymentName, 0)
```
Comment (0xFelix) [inline tests/monitoring/monitoring.go:519]: Why the change of order? Can't `updateScale` just be added to L509?

File: tests/monitoring/monitoring.go [hunk 3]
```diff
@@ -531,42 +538,48 @@ var _ = Describe("[Serial][sig-monitoring]Prometheus Alerts", Serial, func() {
 			waitUntilAlertDoesNotExist(virtHandler.downAlert)
 		})
 
-		It("VirtApiRESTErrorsBurst should be triggered when requests to virt-api are failing", func() {
+		It("VirtApiRESTErrorsBurst and VirtApiRESTErrorsHigh should be triggered when requests to virt-api are failing", func() {
+			randVmName := rand.String(6)
```
Comment (0xFelix) [inline tests/monitoring/monitoring.go:542]: Previously the VM name was changed with every loop iteration but now it stays the same. Is that intentional?

Comment (machadovilaca) [reply tests/monitoring/monitoring.go:542]: yes, just a small optimization, VM does not exist anyway, so no need to regenerate a new random name everytime

**Review by 0xFelix — APPROVED** [review]
LGTM from what I can tell, thanks!

/lgtm

**Comment by machadovilaca** [comment]
@0xFelix can you take a new look? I only rebased the changes

/cc @fossedihelm @xpivarc

**Review by 0xFelix — APPROVED** [review]
/lgtm

File: tests/monitoring/monitoring.go [hunk 4]
```diff
@@ -378,42 +385,48 @@ var _ = Describe("[Serial][sig-monitoring]Monitoring", Serial, decorators.SigMon
...
 			for i := 0; i < 120; i++ {
-				cmd := clientcmd.NewVirtctlCommand("vnc", "test"+rand.String(6))
+				cmd := clientcmd.NewVirtctlCommand("vnc", randVmName)
 				err := cmd.Execute()
 				Expect(err).To(HaveOccurred())
 
 				time.Sleep(500 * time.Millisecond)
```
Comment (0xFelix) [inline tests/monitoring/monitoring.go:395]: Btw. can't we use `Eventually` here?

File: tests/monitoring/monitoring.go [hunk 5]
```diff
@@ -378,83 +385,70 @@ var _ = Describe("[Serial][sig-monitoring]Monitoring", Serial, decorators.SigMon
...
-				if err == nil {
-					return
-				}
+				err1 := checkAlert(virtApi.restErrorsBurtsAlert)
+				err2 := checkAlert(virtApi.restErrorsHighAlert)
+				return err1 == nil && err2 == nil
 			}
 
-			verifyAlertExist(virtApi.restErrorsBurtsAlert)
+			Eventually(triggerAlerts, 5*time.Minute, 500*time.Millisecond).Should(BeTrue())
```
Comment (fossedihelm) [inline tests/monitoring/monitoring.go:401]: I would prefer to create 2 `verifyAlertExists` assertions(one for `VirtApiRESTErrorsBurst` and one for `VirtApiRESTErrorsHigh`): in case of failure we will be able to fast detect which one is failing.
Otherwise, as is, we will be informed that one of them, or both is not true.
If you want to create only one `Eventually` you can do something like:
```
Eventually(func(g Gomega) {
	cmd := clientcmd.NewVirtctlCommand("vnc", randVmName)
	err := cmd.Execute()
	g.Expect(err).To(HaveOccurred())
	g.Expect(checkAlert(virtApi.restErrorsBurtsAlert)).To(BeNil())
	g.Expect(checkAlert(virtApi.restErrorsHighAlert)).To(BeNil())
}, 5*time.Minute, 500*time.Millisecond).Should(Succeed())
```

File: tests/monitoring/monitoring.go [hunk 6]
```diff
@@ -378,83 +385,70 @@ var _ = Describe("[Serial][sig-monitoring]Monitoring", Serial, decorators.SigMon
...
 			Expect(err).ToNot(HaveOccurred())
 
-			for i := 0; i < 60; i++ {
-				time.Sleep(500 * time.Millisecond)
-
-				err := checkAlert(virtOperator.restErrorsBurtsAlert)
-				if err == nil {
-					return
-				}
+			triggerAlerts := func() bool {
```
Comment (fossedihelm) [inline tests/monitoring/monitoring.go:409]: same as above

**Review by fossedihelm — COMMENTED** [review]
Thanks @machadovilaca. Everything seems ok. Just one debug oriented enhancement.

**Comment by fossedihelm** [comment]
/lgtm
Thanks
